File: Iniciante/1018 - Cédulas/Python/1018.py
def main():
    value = int(input())
    print(value)
    banknotes = [100, 50, 20, 10, 5, 2, 1]
    necessary_banknotes(value, banknotes)


def necessary_banknotes(value, banknotes):
    for item in banknotes:
        amount_banknotes = value // item
        print(f"{amount_banknotes} nota(s) de R$ {item},00")
        value %= item


if __name__ == "__main__":
    main()

File: Iniciante/1018 - Cédulas/TypeScript/1018.ts
import * as fs from "fs";

function main(): void {
  const input: string = fs.readFileSync("/dev/stdin", "utf8");
  const value: number = parseInt(input);
  console.log(value);
  const banknotes: number[] = [100, 50, 20, 10, 5, 2, 1];
  necessaryBanknotes(banknotes, value);
}

function necessaryBanknotes(banknotes: number[], value: number): void {
  for (let i = 0; i < banknotes.length; i++) {
    let amountBanknotes: number = Math.trunc(value / banknotes[i]);
    console.log(`${amountBanknotes} nota(s) de R$ ${banknotes[i]},00`);
    value %= banknotes[i];
  }
}

main();
